packages/platforms/inventec/x86-64/x86-64-inventec-d7054q28b/onlp/builds/src/module/src/sysi.c

```diff
@@ -0,0 +1,295 @@
+
+    /* 4 Thermal sensors on the chassis */
+    for (i = 1; i <= NUM_OF_THERMAL_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* 5 LEDs on the chassis */
+    for (i = 1; i <= NUM_OF_LED_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 2 PSUs on the chassis */
+    for (i = 1; i <= NUM_OF_PSU_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    /* 4 Fans on the chassis */
+    for (i = 1; i <= NUM_OF_FAN_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+typedef struct fan_ctrl_policy {
+   int duty_cycle;
+   int temp_down_adjust; /* The boundary temperature to down adjust fan speed */
+   int temp_up_adjust;   /* The boundary temperature to up adjust fan speed */
+} fan_ctrl_policy_t;
+
+fan_ctrl_policy_t  fan_ctrl_policy_f2b[] = {
+{32,      0, 174000},
+{38, 170000, 182000},
+{50, 178000, 190000},
+{63, 186000,      0}
+};
+
+fan_ctrl_policy_t  fan_ctrl_policy_b2f[] = {
+{32,     0,  140000},
+{38, 135000, 150000},
+{50, 145000, 160000},
+{69, 155000,      0}
+};
+
+#define FAN_DUTY_CYCLE_MAX  100
+#define FAN_SPEED_CTRL_PATH "/sys/bus/i2c/devices/2-0066/fan_duty_cycle_percentage"
+
+/*
+ * For AC power Front to Back :
+ *	* If any fan fail, please fan speed register to 15
+ *	* The max value of Fan speed register is 9
+ *		[LM75(48) + LM75(49) + LM75(4A)] > 174  => set Fan speed value from 4 to 5
+ *		[LM75(48) + LM75(49) + LM75(4A)] > 182  => set Fan speed value from 5 to 7
+ *		[LM75(48) + LM75(49) + LM75(4A)] > 190  => set Fan speed value from 7 to 9
+ *
+ *		[LM75(48) + LM75(49) + LM75(4A)] < 170  => set Fan speed value from 5 to 4
+ *		[LM75(48) + LM75(49) + LM75(4A)] < 178  => set Fan speed value from 7 to 5
+ *		[LM75(48) + LM75(49) + LM75(4A)] < 186  => set Fan speed value from 9 to 7
+ *
+ *
+ * For  AC power Back to Front :
+ *	* If any fan fail, please fan speed register to 15
+ *	* The max value of Fan speed register is 10
+ *		[LM75(48) + LM75(49) + LM75(4A)] > 140  => set Fan speed value from 4 to 5
+ *		[LM75(48) + LM75(49) + LM75(4A)] > 150  => set Fan speed value from 5 to 7
+ *		[LM75(48) + LM75(49) + LM75(4A)] > 160  => set Fan speed value from 7 to 10
+ *
+ *		[LM75(48) + LM75(49) + LM75(4A)] < 135  => set Fan speed value from 5 to 4
+ *		[LM75(48) + LM75(49) + LM75(4A)] < 145  => set Fan speed value from 7 to 5
+ *		[LM75(48) + LM75(49) + LM75(4A)] < 155  => set Fan speed value from 10 to 7
+ */
+int
+onlp_sysi_platform_manage_fans(void)
+{
+    int i = 0, arr_size, temp;
+    fan_ctrl_policy_t *policy;
+    int cur_duty_cycle, new_duty_cycle;
+    onlp_thermal_info_t thermal_1, thermal_2, thermal_3;
+
+    int  fd, len;
+    char  buf[10] = {0};
+
+    /* Get each fan status
+     */
+    for (i = 1; i <= NUM_OF_FAN_ON_MAIN_BROAD; i++)
+    {
+        onlp_fan_info_t fan_info;
+
+        if (onlp_fani_info_get(ONLP_FAN_ID_CREATE(i), &fan_info) != ONLP_STATUS_OK) {
```

Comment:
This must call onlp_fan_info_get, not onlp_fani_info_get(). 
This function does not run under the API lock.